docs/pipeline/triggering/DRIFT_TRIGGERS.md

```diff
@@ -0,0 +1,231 @@
+        +void init_detector()
+        +dict[MetricId, MetricResult] detect_drift(embeddings_ref, embeddings_cur, bool is_warmup)
+    }
+
+    class AlibiDriftDetector
+    class EvidentlyDriftDetector
+    class BaseMetric {
+        decision_criterion: DecisionCriterion
+    }
+
+    DriftDetector <|-- AlibiDriftDetector
+    DriftDetector <|-- EvidentlyDriftDetector
+
+
+    AlibiDriftDetector *-- "|metrics|" AlibiDetectDriftMetric
+    EvidentlyDriftDetector *-- "|metrics|" EvidentlyDriftMetric
+
+    class AlibiDetectDriftMetric {
+        <<abstract>>
+    }
+
+    class AlibiDetectMmdDriftMetric {
+    }
+
+    class AlibiDetectCVMDriftMetric {
+    }
+
+    class AlibiDetectKSDriftMetric {
+    }
+
+    BaseMetric <|-- AlibiDetectDriftMetric
+    AlibiDetectDriftMetric <|-- AlibiDetectMmdDriftMetric
+    AlibiDetectDriftMetric <|-- AlibiDetectCVMDriftMetric
+    AlibiDetectDriftMetric <|-- AlibiDetectKSDriftMetric
+
+    class EvidentlyDriftMetric {
+        <<abstract>>
+        int num_pca_component
+    }
+
+    class EvidentlyModelDriftMetric {
+        bool bootstrap = False
+        float quantile_probability = 0.95
+        float threshold = 0.55
+    }
+
+    class EvidentlyRatioDriftMetric {
+        string component_stattest = "wasserstein"
+        float component_stattest_threshold = 0.1
+        float threshold = 0.2
+    }
+
+    class EvidentlySimpleDistanceDriftMetric {
+        string distance_metric = "euclidean"
+        bool bootstrap = False
+        float quantile_probability = 0.95
+        float threshold = 0.2
+    }
+
+    BaseMetric <|-- EvidentlyDriftMetric
+    EvidentlyDriftMetric <|-- EvidentlyModelDriftMetric
+    EvidentlyDriftMetric <|-- EvidentlyRatioDriftMetric
+    EvidentlyDriftMetric <|-- EvidentlySimpleDistanceDriftMetric
+
+```
+
+### DecisionCriterion Hierarchy
+
+The `DecisionCriterion` class is an abstract configuration base class for criteria like `ThresholdDecisionCriterion` and `DynamicThresholdCriterion`, which define how decisions are made based on drift metrics.
+
+```mermaid
+classDiagram
+    class DecisionCriterion {
+        <<abstract>>
+    }
+
+    class ThresholdDecisionCriterion {
+        float threshold
+    }
+
+    class DynamicThresholdCriterion {
+        int window_size = 10
+        float percentile = 0.05
+    }
+
+    DecisionCriterion <|-- ThresholdDecisionCriterion
+    DecisionCriterion <|-- DynamicThresholdCriterion
+```
+
+### DriftDecisionPolicy Hierarchy
+
+The `DriftDecisionPolicy` class is an abstract base class for policies like `ThresholdDecisionPolicy`, `DynamicDecisionPolicy`, and `HypothesisTestDecisionPolicy`.
+
+Each Decision policy wraps one DriftMetric (e.g. MMD, CVM, KS) and one DecisionCriterion (e.g. Threshold, DynamicThreshold, HypothesisTest) to make a decision based on the distance metric. It e.g. observes the series of distance value measurements from it's `DriftMetric` and makes a decision after having calibrated on the seen distances.
```

Comment:
Should we explain the terminology we use in context of the dynamic policy somewhere? For a new reader of the drift trigger file, the terms calibiration and warmup might still be confusing, might be good to define them somewhere / explain the calibration logic